app/services/activitypub/process_account_service.rb

```diff
@@ -117,6 +117,8 @@ def set_immediate_attributes!
     @account.indexable               = @json['indexable'] || false
     @account.memorial                = @json['memorial'] || false
     @account.attribution_domains     = as_array(@json['attributionDomains'] || []).map { |item| value_or_id(item) }
+    @account.avatar_description      = @json['avatarDescription'] || ''
+    @account.header_description      = @json['headerDescription'] || ''
```

Comment:
This isn't quite right, remember the ActivityStreams 2 vocabulary page I showed you & the `image_url` method? (line 200/204)

essentially you need to look inside the `icon` and `image` properties to discover the `summary` or `name` properties.

Reply:
You'll also want to make use of `.presence?` here (might be without the question mark, I can't remember)

Reply:
oohh, ok, I think I get this now